lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -655,6 +657,32 @@ def test_html_view_for_invalid_certificate(self):
         self.assertIn("Cannot Find Certificate", response.content)
         self.assertIn("We cannot find a certificate with this URL or ID number.", response.content)
 
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    def test_html_lang_attribute_is_dynamic_for_invalid_certificate_html_view(self):
+        """
+        Tests that Certificate HTML Web View's lang attribute is based on user language.
+        """
+        self._add_course_certificates(count=1, signatory_count=2)
+        test_url = get_certificate_url(
+            user_id=self.user.id,
+            course_id=unicode(self.course.id)
+        )
+
+        # invalidate certificate and verify that "Cannot Find Certificate" is returned
+        self.cert.invalidate()
+
+        user_language = 'fr'
+        translation.activate(user_language)
+        self.client.cookies[settings.LANGUAGE_COOKIE] = user_language
```

Comment:
Why do you need to set the cookie if you've already changed the Django language to `user_language` using `translation.activate`/`translation.override`?

Reply:
@amangano-edx translation.override is not working correctly for above test case. The language does get activated but the change doesn't last for client.get() call. I initially tried below snippet but no success
```
 with translation.override('fr'):
            self.assertEqual(get_language(), 'fr')
            response = self.client.get(test_url)
            self.assertIn('<html class="no-js" lang="fr">', response.content)
``` 
The first assert passes, but inspecting response.content shows the default lang.  translation.activate('user_language') only loads, but does't activates/switch in my case, so i need to either set the cookie or accept-header params to activate this.  Inspecting client.cookies dictionary in test showed "openedx-language-preference" key was empty. Setting this value causes django to activate language based on its value. I guess we are requesting a view so django evaluates language based on these rules. https://docs.djangoproject.com/en/2.0/topics/i18n/translation/#how-django-discovers-language-preference. Have removed translation.activation as it was not working at all.